Library/Homebrew/diagnostic.rb

```diff
@@ -443,6 +443,32 @@ def check_access_cellar
         EOS
       end
 
+      def check_for_empty_taps
+        taps_dir = HOMEBREW_LIBRARY/"Taps"
+        return unless taps_dir.exist?
+        return unless taps_dir.directory?
+
+        found_empty_dirs = []
+
+        # Warn USER of empty taps_dir children directories
+        taps_dir.children.select{ |c| c.directory? } \
```

Comment:
Could you explain this `Tap.map(&:path)` snippet to me? I'm not familiar with this and still getting used to ruby syntax.